### PR TITLE
Timeshift API: Add support for GetBufferTimeStart() and GetBufferTimeEnd()

### PR DESCRIPTION
**Pull request by ksooo**

### CODE REVIEW

**Comment by FernetMenta** [comment]
I don't know much about tvh but I want to give you some info about something I noticed while I did vnsi. GetBufferTimeStart() and friends are called from a different thread than demuxer methods

File: src/Tvheadend.cpp [hunk 1]
```diff
@@ -2283,3 +2283,13 @@ time_t CTvheadend::DemuxGetTimeshiftTime() const
 {
   return m_dmx_active->GetTimeshiftTime();
 }
+
+time_t CTvheadend::DemuxGetTimeshiftBufferStart() const
+{
+  return m_dmx_active.GetTimeshiftBufferStart();
```
Comment (Jalle19) [inline src/Tvheadend.cpp]: `m_dmx_active` is a pointer, at least judging by the method above this one.

File: src/Tvheadend.h [hunk 2]
```diff
@@ -242,6 +242,16 @@ class CHTSPDemuxer
   {
     return (time_t)m_timeshiftStatus.shift;
   }
+  inline time_t GetTimeshiftBufferStart() const
+  {
+    // Note: start/end mismatch is not a bug. tvh uses inversed naming logic here!
+    return static_cast<time_t>(m_timeshiftStatus.end);
```
Comment (Jalle19) [inline src/Tvheadend.h]: I'm not sure casting to `time_t` is correct here. While working on some refactoring locally I changed the method above these to return `int64_t` instead. Considering tvheadend reports these values as microseconds it's quite easy to end up truncating the value if casting to a 32-bit value at this stage.

Comment (ksooo) [inline src/Tvheadend.h]: Makes sense. Will change this. Thx.

File: src/Tvheadend.cpp (hunk 1 above)
Comment (ksooo) [inline src/Tvheadend.cpp]: Damn!

File: src/Tvheadend.h (hunk 2 above)
Comment (Jalle19) [inline src/Tvheadend.h]: TBH I have no idea what I was thinking.

**Comment by ksooo** [comment]
@Jalle19 Fixed. Better now?

**Comment by Jalle19** [comment]
Looks reasonable

**Comment by ksooo** [comment]
Took the chance to fix the two Coverity issues introduced by predictive tuning (as a separate commit)...

**Comment by Jalle19** [comment]
Thanks!

**Pull request merged**